x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -369,59 +363,72 @@ export class TaskManagerRunner implements TaskRunner {
     }
   }
 
-  private validateTaskParams({ taskInstance }: RunContext) {
-    let error;
+  private async validateTask({ taskInstance }: RunContext) {
     const { state, taskType, params, id, numSkippedRuns = 0 } = taskInstance;
+
+    if (!this.requeueInvalidTasksConfig.enabled) {
+      return { state };
+    }
+
     const { max_attempts: maxAttempts } = this.requeueInvalidTasksConfig;
+    const { paramsSchema, indirectParamsSchema, latestTypeVersion } = this.definition;
+    let hasValidationError = false;
 
-    try {
-      const paramsSchema = this.definition.paramsSchema;
-      if (paramsSchema) {
+    // validate task params
+    if (paramsSchema) {
+      try {
         paramsSchema.validate(params);
-      }
-    } catch (err) {
-      this.logger.warn(`Task (${taskType}/${id}) has a validation error: ${err.message}`);
-      if (numSkippedRuns < maxAttempts) {
-        error = createSkipError(err);
-      } else {
-        this.logger.warn(
-          `Task Manager has reached the max skip attempts for task ${taskType}/${id}`
-        );
+      } catch (err) {
+        hasValidationError = true;
+        this.logger.warn(`Task (${taskType}/${id}) has a validation error: ${err.message}`);
+        if (numSkippedRuns < maxAttempts) {
+          return { error: createSkipError(err), state };
+        }
       }
     }
 
-    return { ...(error ? { error } : {}), state };
-  }
+    if (this.task?.loadIndirectParams) {
+      const { data } = await this.task.loadIndirectParams();
+      const indirectParams = data?.indirectParams;
+      const typeVersion = data?.typeVersion;
 
-  private async validateIndirectTaskParams({ taskInstance }: RunContext) {
-    let error;
-    const { state, taskType, id, numSkippedRuns = 0 } = taskInstance;
-    const { max_attempts: maxAttempts } = this.requeueInvalidTasksConfig;
-    const indirectParamsSchema = this.definition.indirectParamsSchema;
+      // validate runtime version
+      if (typeVersion && latestTypeVersion && typeVersion > latestTypeVersion) {
+        hasValidationError = true;
+        this.logger.warn(
+          `Task (${taskType}/${id}) has a newer version(${typeVersion}) than expected((${latestTypeVersion}))`
+        );
+        if (numSkippedRuns < maxAttempts) {
+          return {
+            error: createSkipError(
+              new Error('Task has already been run by a newer Kibana version')
+            ),
+            state,
+          };
+        }
+      }
 
-    if (this.task?.loadIndirectParams && !!indirectParamsSchema) {
-      const { data } = await this.task.loadIndirectParams();
-      if (data) {
+      // validate indirect params e.g. connector or rule.params
+      if (indirectParamsSchema && indirectParams) {
```

Comment:
Patrick left a note about this check in this PR's POC.
When there is a version mismatch the task is skipped 100 times (default number) then doesn't return a skipError in order to let the rule run.
So it comes to this line and does `indirectParamsSchema` validation as well. As it has already reached the `maxAttempts` just logs a warning and doesn't return any skipError either.

We may skip this validation when there is already an error (hasValidationError). WDYT?